include/deal.II/base/vectorization.h

```diff
@@ -165,16 +167,41 @@ class VectorizedArrayIterator
   VectorizedArrayIterator<T> &
   operator++()
   {
-    lane++;
+    AssertIndexRange(lane + 1, T::size() + 1) lane++;
+    return *this;
+  }
+
+  /**
+   * This operator advances the iterator by @p offet lanes and returns a
+   * reference to <tt>*this</tt>.
+   */
+  VectorizedArrayIterator<T> &
+  operator+=(const unsigned int offset)
+  {
+    AssertIndexRange(lane + offset, T::size() + 1) lane += offset;
+    return *this;
+  }
+
+  /**
+   * Prefix <tt>--</tt> operator: <tt>--iterator</tt>. This operator advances
+   * the iterator to the previous lane and returns a reference to
+   * <tt>*this</tt>.
+   */
+  VectorizedArrayIterator<T> &
+  operator--()
+  {
+    AssertIndexRange(lane, T::size() + 1);
```

Comment:
This is wrong. It should have been
```suggestion
    Assert(lane>0, ExcMessage ("You can't decrement an iterator that is already at the beginning of the range.");
```